packages/sdk/src/evm/common/plugin.ts

```diff
@@ -105,6 +110,61 @@ async function getPluginABI(
   ).map((metadata) => metadata.abi);
 }
 
+export async function getPluginNames(
+  publishedMetadataUri: string,
+  storage: ThirdwebStorage,
+): Promise<string[]> {
+  const extendedMetadata = await fetchExtendedReleaseMetadata(
+    publishedMetadataUri,
+    storage,
+  );
+
+  let contractNames: string[] = [];
+
+  if (extendedMetadata.factoryDeploymentData?.implementationAddresses) {
+    const implementationsAddresses = Object.entries(
+      extendedMetadata.factoryDeploymentData.implementationAddresses,
+    );
+
+    try {
+      for (const [network, implementation] of implementationsAddresses) {
```

Comment:
instead of a for loop, you can do 

`const implementation = implementationsAddresses.find((entry) => entry.implementation !== "")`

will return the first one that  matches your condition (or undefined if none are found)